discovery-infra/assisted_service_api.py

```diff
@@ -14,9 +18,52 @@ def __init__(self, inventory_url):
         self.inventory_url = inventory_url
         configs = Configuration()
         configs.host = self.inventory_url + "/api/assisted-install/v1"
+        self.set_config_auth(configs)
+
         self.api = ApiClient(configuration=configs)
         self.client = api.InstallerApi(api_client=self.api)
 
+    def set_config_auth(self, c):
+        offline_token = os.environ.get('OFFLINE_TOKEN', "")
+        if offline_token == "": return
+
+        def refresh_api_key(config):
+            # Get the properly padded key segment
+            auth = config.api_key.get('Authorization', None)
+            if auth != None:
+                segment = auth.split('.')[1]
+                padding = len(segment) % 4
+                segment = segment + padding * '='
+
+                expires_on = json.loads(base64.b64decode(segment))['exp']
+
+                # if this key doesn't expire or if it has more than 30 seconds left, don't refresh
+                remaining = expires_on - time.time()
+                if expires_on == 0 or remaining > 30:
+                    return
+
+            # fetch new key if expired or not set yet
+            token_url = "https://sso.redhat.com/auth/realms/redhat-external/protocol/openid-connect/token"
+            env_url = os.environ.get("SSO_URL", "")
```

Comment:
token_url = os.environ.get("SSO_URL", "https://sso.redhat.com/auth/realms/redhat-external/protocol/openid-connect/token")

Reply:
u can set default SSO_URL in Makefile

Reply:
i think just set it default

Reply:
Yeah, setting the default in the Makefile would work for this one. Generally the weirdness around how I'm checking for env vars (and checking for empty strings specifically) is because of how skipper automatically sets env vars in `skipper.yaml` to `""` if they were unset by the user.

I call this out in the commit message of 9b7414aa495f9c6c91eb6f10816f7615a3948576.